src/nvim/tui/driver-csi.c

```diff
@@ -131,17 +130,16 @@ static TermKeyResult handle_csifunc(TermKey *tk, TermKeyKey *key, int cmd, TermK
     int mod = key->modifiers;
     (*tk->method.emit_codepoint)(tk, args[2], key);
     key->modifiers |= mod;
-  }
-  else if(args[0] >= 0 && args[0] < NCSIFUNCS) {
+  } else if (args[0] >= 0 && (size_t)args[0] < NCSIFUNCS) {
```

Comment:
This cast is not needed anymore
```suggestion
  } else if (args[0] >= 0 && args[0] < NCSIFUNCS) {
```